app/models/room.rb

```diff
@@ -11,7 +11,7 @@ def to_param
   end
 
   def self.random_room
-    self.limit(1).order("RANDOM()").take
+    self.joins(:jams).where("jams.id IS NOT NULL").limit(1).order("RANDOM()").take
```

Comment:
I'd go with a scope (and slightly different name) so you can `.take` one (side not, no `limit(1)` is needed since that's all `.take` adds) or `.limit` many. Like so:
```
scope :recommended, -> { joins(:jams).order("RANDOM()") }
``` (side note: joins already makes sure any jams is attached, no need adding `where`)

usage `Room.recommended.take`